src/test/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapperTest.java

```diff
@@ -0,0 +1,99 @@
+package com.icthh.xm.uaa.security.ldap;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.icthh.xm.uaa.domain.User;
+import com.icthh.xm.uaa.domain.properties.TenantProperties;
+import com.icthh.xm.uaa.security.DomainUserDetailsService;
+import com.icthh.xm.uaa.service.UserService;
+import com.icthh.xm.uaa.service.dto.UserDTO;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.ldap.core.DirContextOperations;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UaaLdapUserDetailsContextMapperTest {
+
+    private static final String LOGIN = "Homer";
+    public static final Map<String, String> GROUP_MAPPING = Map.of(
+        "GROUP1", "SUPER_ADMIN",
+        "GROUP2", "LOSER"
+    );
+    @Mock
+    UserService userService;
+
+    @Mock
+    DirContextOperations ctx;
+
+    @Mock
+    TenantProperties.Ldap ldapConf;
+
+    @Mock
+    DomainUserDetailsService userDetailsService;
+
+    @InjectMocks
+    UaaLdapUserDetailsContextMapper uaaLdapUserDetailsContextMapper;
+
+    @Test
+    public void userRoleMustNotChangeDuringLogin() {
+        whenFindUserByLogin();
+        when(ldapConf.getRole()).thenReturn(new TenantProperties.Ldap.Role("DEFAULT_ROLE", null));
+        uaaLdapUserDetailsContextMapper.mapUserFromContext(ctx, LOGIN, List.of());
+        verify(userDetailsService).loadUserByUsername(eq(LOGIN));
+        verify(userService, never()).saveUser(any(User.class));
```

Comment:
is it enough to verify that we never save the user or better to make a specific check that role still the same?

Reply:
added additional check